addons/stock/models/stock_move_line.py

```diff
@@ -521,7 +521,7 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
             move_to_recompute_state = self.env['stock.move']
 
             rounding = self.product_uom_id.rounding
-            for candidate in oudated_candidates:
+            for candidate in reversed(oudated_candidates.sorted(lambda ml: ml.move_id.date_expected)):
```

Comment:
1. The method `sorted()` may get a keyword `reverse=True`, so there is no need to use Python's `reversed`
1. AFAIK, even the lambda is not required, it would be as simple as:
   `for candidate in oudated_candidates.sorted('move_id.date_expected', reverse=True):`

Reply:
Thanks for the review.
Ad 1. True. Will update.
Ad 2. I guess I validated that. I'll validate again.